haystack/nodes/file_converter/parsr.py

```diff
@@ -290,5 +311,17 @@ def _convert_table_element(
         if self.add_page_number:
             table_meta["page"] = page_idx + 1
 
+        if extract_headlines:
+            relevant_headlines = []
+            cur_lowest_headline_level = 1000
+            for headline in reversed(headlines):
+                if headline["level"] < cur_lowest_headline_level:
+                    headline_copy = copy.deepcopy(headline)
+                    headline_copy["start_idx"] = None
+                    relevant_headlines.append(headline_copy)
+                    cur_lowest_headline_level = headline_copy["level"]
+            relevant_headlines = list(reversed(relevant_headlines))
```

Comment:
nit: the "alien face" operator would be faster here: `relevant_headlines = relevant_headlines[::-1]`. I expect the list to be small enough for this not not matter, leaving here just in case.